tool/extract_all_pub_dev.dart

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2020, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// This is a manual test that can be run to test the .tar.gz decoding.
+/// It will save progress in [statusFileName] such that it doesn't have to be
+/// finished in a single run.
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:http/http.dart' as http;
+import 'package:pool/pool.dart';
+import 'package:pub/src/http.dart';
+import 'package:pub/src/io.dart';
+
+const statusFilename = 'extract_all_pub_status.json';
+
+Stream<String> allPackageNames() async* {
+  var nextUrl = 'https://pub.dev/api/packages';
```

Comment:
try `https://pub.dartlang.org/api/packages?compact=1` then you'll get all package names in one request.

Reply:
Ah - nice.